src/frontend/src/handler/fetch_cursor.rs

```diff
@@ -12,21 +12,98 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use pgwire::pg_field_descriptor::PgFieldDescriptor;
 use pgwire::pg_response::{PgResponse, StatementType};
-use risingwave_sqlparser::ast::ObjectName;
+use pgwire::types::Row;
+use risingwave_sqlparser::ast::{FetchCursorStatement, ObjectName, Statement};
 
+use super::declare_cursor::create_stream_for_cursor;
+use super::util::gen_query_from_logstore_ge_rw_timestamp;
 use super::RwPgResponse;
-use crate::error::Result;
+use crate::error::{ErrorCode, Result};
 use crate::handler::HandlerArgs;
+use crate::session::cursor_manager::CursorRowValue;
+use crate::{Binder, PgResponseStream};
 
 pub async fn handle_fetch_cursor(
-    handler_args: HandlerArgs,
-    cursor_name: ObjectName,
-    count: Option<i32>,
+    handle_args: HandlerArgs,
+    stmt: FetchCursorStatement,
 ) -> Result<RwPgResponse> {
-    let session = handler_args.session;
-    let (rows, pg_descs) = session.cursor_next(&cursor_name, count).await?;
-    Ok(PgResponse::builder(StatementType::FETCH_CURSOR)
-        .values(rows.into(), pg_descs)
-        .into())
+    let session = handle_args.session.clone();
+    let db_name = session.database();
+    let (schema_name, cursor_name) =
+        Binder::resolve_schema_qualified_name(db_name, stmt.cursor_name.clone())?;
+
+    let cursor_manager = session.get_cursor_manager();
+
+    let (cursors, pg_descs) = cursor_manager
+        .get_row_with_cursor(cursor_name.clone(), stmt.count)
+        .await?;
+    let (rows, next_query_parameter) = covert_cursor_row_values(cursors);
+
+    match next_query_parameter {
```

Comment:
Given that we already have a enum for `Cursor`, can we put these logics in `cursor.next()`? They seems to be implementation details for cursor.